service/controller/resource/ipam/azure_machinepool_checker.go

```diff
@@ -0,0 +1,80 @@
+package ipam
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	capzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	"sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type AzureMachinePoolCheckerConfig struct {
+	CtrlClient client.Client
+	Logger     micrologger.Logger
+}
+
+type AzureMachinePoolChecker struct {
```

Comment:
We could call this one `AzureMachinePoolSubnetChecker`, so it's more explicit and obvious what it is about. Also analogously `AzureMachinePoolPersister` -> `AzureMachinePoolSubnetPersister`. Unless there are some other use cases and these will be checking and persisting other things besides subnet in the future.

WDYT @tuommaki?